src/stores/room-list-v3/RoomListStoreV3.ts

```diff
@@ -78,7 +83,93 @@ export class RoomListStoreV3Class extends AsyncStoreWithClient<EmptyObject> {
     }
 
     protected async onAction(payload: ActionPayload): Promise<void> {
-        return;
+        if (!this.matrixClient || !this.roomSkipList?.initialized) return;
+
+        /**
+         * For the kind of updates that we care about (represented by the cases below),
+         * we try to find the associated room and simply re-insert it into the
+         * skiplist. If the position of said room in the sorted list changed, re-inserting
+         * would put it in the correct place.
+         */
+        switch (payload.action) {
+            case "MatrixActions.Room.receipt": {
+                if (readReceiptChangeIsFor(payload.event, this.matrixClient)) {
+                    const room = payload.room;
+                    if (!room) {
+                        logger.warn(`Own read receipt was in unknown room ${room.roomId}`);
+                        return;
+                    }
+                    this.addRoomAndEmit(room);
+                }
+                break;
+            }
+
+            case "MatrixActions.Room.tags": {
+                const room = payload.room;
+                this.addRoomAndEmit(room);
+                break;
+            }
+
+            case "MatrixActions.Event.decrypted": {
+                const roomId = payload.event.getRoomId();
+                if (!roomId) return;
+                const room = this.matrixClient.getRoom(roomId);
+                if (!room) {
+                    logger.warn(`Event ${payload.event.getId()} was decrypted in an unknown room ${roomId}`);
+                    return;
+                }
+                this.addRoomAndEmit(room);
+                break;
+            }
+
+            case "MatrixActions.accountData": {
+                if (payload.event_type !== EventType.Direct) return;
+                const dmMap = payload.event.getContent();
+                for (const userId of Object.keys(dmMap)) {
+                    const roomIds = dmMap[userId];
+                    for (const roomId of roomIds) {
+                        const room = this.matrixClient.getRoom(roomId);
+                        if (!room) {
+                            logger.warn(`${roomId} was found in DMs but the room is not in the store`);
+                            continue;
+                        }
+                        this.addRoomAndEmit(room);
+                    }
+                }
+                break;
```

Comment:
This will fire `LISTS_UPDATE_EVENT` multiple times. Can we add the rooms to the skip list and after fire once the event?

Reply:
Done in https://github.com/element-hq/element-web/pull/29397/commits/bab8d937bc319794e39afa857c5d7991c65af816